src/display/network.js

```diff
@@ -351,51 +351,24 @@ function PDFNetworkStreamFullRequestReader(manager, options) {
 }
 
 PDFNetworkStreamFullRequestReader.prototype = {
-  _validateRangeRequestCapabilities: function
-      PDFNetworkStreamFullRequestReader_validateRangeRequestCapabilities() {
+  getResponseHeader(name) {
+    let fullRequestXhrId = this._fullRequestId;
+    let fullRequestXhr = this._manager.getRequestXhr(fullRequestXhrId);
 
-    if (this._disableRange) {
-      return false;
-    }
-
-    var networkManager = this._manager;
-    if (!networkManager.isHttp) {
-      return false;
-    }
-    var fullRequestXhrId = this._fullRequestId;
-    var fullRequestXhr = networkManager.getRequestXhr(fullRequestXhrId);
-    if (fullRequestXhr.getResponseHeader('Accept-Ranges') !== 'bytes') {
-      return false;
-    }
-
-    var contentEncoding =
-      fullRequestXhr.getResponseHeader('Content-Encoding') || 'identity';
-    if (contentEncoding !== 'identity') {
-      return false;
-    }
-
-    var length = fullRequestXhr.getResponseHeader('Content-Length');
-    length = parseInt(length, 10);
-    if (!isInt(length)) {
-      return false;
-    }
-
-    this._contentLength = length; // setting right content length
-
-    if (length <= 2 * this._rangeChunkSize) {
-      // The file size is smaller than the size of two chunks, so it does
-      // not make any sense to abort the request and retry with a range
-      // request.
-      return false;
-    }
-
-    return true;
+    return fullRequestXhr.getResponseHeader(name);
   },
 
   _onHeadersReceived:
       function PDFNetworkStreamFullRequestReader_onHeadersReceived() {
-
-    if (this._validateRangeRequestCapabilities()) {
+    // Setting right content length.
+    this._contentLength =
+      parseInt(this.getResponseHeader('Content-Length'), 10);
```

Comment:
Won't just assigning a value, without doing any validation, mean that if the Content-Length header isn't present then `this._contentLength` will be `NaN` which probably isn't very good (compare with the old code which uses a `!isInt(...)` check)?

*The following might not matter, but asking anyway since I'm not sure:*
With this patch, you're always setting `this._contentLength` whereas before that was only done when range requests were supported; does this matter at all?